massmigration/migrations.py

```diff
@@ -33,14 +47,57 @@ class BaseMigration:
     # this attribute.
     backend_method: str = None
 
-    def __init__(self, app_label, name):
+    # We always allow the migration to run without specifying a using. Letting Django (and Django router) do its course.
+    # If None, the migration can be run on all the available databases.
+    allowed_database_aliases = None
+
+    migration_record_db_alias = None
+
+    @classmethod
+    def get_allowed_databases(cls):
+        """
+        Returns a list of allowed database aliases for running the
+        migration assuming that letting Django (and Django router) do its
+        course is always an allowed option.
+        If `allowed_database_aliases` is None, the migration can be run on
+        all databases.
+        """
+
+        if cls.allowed_database_aliases is None:  # If None we assume can run on all DBs
+            available_dbs = _get_valid_db_aliases()
+            if len(available_dbs) > 1:  # If we only have one DB there's no need to add it, since we alread have the None
```

Comment:
If I'm reading the code correctly, then if this `if` condition is false, we'll get an undefined variable at the `return` statement?